File: Services/Igmite.Lighthouse.Models/Common/DropdownModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;

namespace Igmite.Lighthouse.Models
{
    [DataContract, Serializable]
    public class DropdownModel<T>
    {
        [Key, DataMember]
        public T Id { get; set; }

        [Key, DataMember]
        public string Name { get; set; }

        [DataMember]
        public string Description { get; set; }

        [DataMember]
        public bool IsSelected { get; set; }

        [DataMember]
        public bool IsDisabled { get; set; }

        [DataMember]
        public int SequenceNo { get; set; }
    }
}

File: Web/src/app/models/dropdown.model.ts

export class DropdownModel {
    Id: string;
    Name: string;
    Description: string;
    IsSelected: boolean;
    IsDisabled: boolean;
    SequenceNo: number;

    constructor(dropdownModelItem?: any) {
        dropdownModelItem = dropdownModelItem || {};

        this.Id = dropdownModelItem.Id || "";
        this.Name = dropdownModelItem.Name || "";
        this.Description = dropdownModelItem.Description || "";
        this.IsSelected = dropdownModelItem.IsSelected || false;
        this.IsDisabled = dropdownModelItem.IsDisabled || false;
        this.SequenceNo = dropdownModelItem.SequenceNo || 1;
    }
}
